pkgs/ffi/test/utf16_test.dart

```diff
@@ -3,26 +3,25 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:ffi';
-import 'dart:typed_data';
 
 import 'package:ffi/ffi.dart';
 import 'package:test/test.dart';
 
 void main() {
   test('toUtf16 ASCII', () {
-    final String start = 'Hello World!\n';
-    final Pointer<Uint16> converted = start.toNativeUtf16().cast();
-    final Uint16List end = converted.asTypedList(start.codeUnits.length + 1);
+    final start = 'Hello World!\n';
+    final converted = start.toNativeUtf16().cast() as Pointer<Uint16>;
```

Comment:
`cast` itself gets a type argument, we shouldn't be using `as` here.

Reply:
The question is why `final foo = bar.cast()` does not fail? I think it should, then this error wouldn't have happend.

Regarding readability, I understand that `cast` gets the type parameter from the definition, but in order to understand what is happening I need to go back to beginning of the line. Also Dart type casts via `as` have the type right there: `final bar = foo as Bar` not `Bar bar = foo as`.